extensions/websockets-next/server/runtime/src/main/java/io/quarkus/websockets/next/runtime/ConnectionManager.java

```diff
@@ -1,27 +1,40 @@
 package io.quarkus.websockets.next.runtime;
 
+import java.util.Iterator;
 import java.util.List;
 import java.util.Set;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentMap;
 import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.stream.Stream;
 
 import jakarta.annotation.PreDestroy;
 import jakarta.inject.Singleton;
 
 import org.jboss.logging.Logger;
 
+import io.quarkus.websockets.next.OpenConnections;
 import io.quarkus.websockets.next.WebSocketConnection;
 
 @Singleton
-public class ConnectionManager {
+public class ConnectionManager implements OpenConnections {
 
     private static final Logger LOG = Logger.getLogger(ConnectionManager.class);
 
     private final ConcurrentMap<String, Set<WebSocketConnection>> endpointToConnections = new ConcurrentHashMap<>();
 
     private final List<ConnectionListener> listeners = new CopyOnWriteArrayList<>();
 
+    @Override
+    public Iterator<WebSocketConnection> iterator() {
+        return stream().iterator();
```

Comment:
This can be moved as a default method in the `OpenConnections` interface

Reply:
It could but for some reason I don't think it's appropriate. Let's keep it here for now...